react/DisableBodyScroll/DisableBodyScroll.js

```diff
@@ -0,0 +1,35 @@
+import PropTypes from 'prop-types';
+import withSideEffect from 'react-side-effect';
+
+const DisableBodyScroll = props => props.children;
+
+DisableBodyScroll.defaultProps = {
+  isDisabled: false
+};
+
+DisableBodyScroll.propTypes = {
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node
+  ]),
+  isDisabled: PropTypes.bool
+};
+
+const reducePropsToState = propsList => {
+  let isDisabled;
+
+  propsList.forEach(props => {
+    isDisabled = props.isDisabled;
+  });
+
+  return isDisabled;
+};
+
+const handleStateChangeOnClient = isDisabled => {
+  Object.assign(document.body.style, { overflow: isDisabled ? 'hidden' : 'auto' });
```

Comment:
Could be better to hoist this ternary higher and only apply `overflow: hidden` if `isDisabled` is true and do nothing if false rather than explicitly setting it to `auto`. Just in case it was being set differently elsewhere, makes it less opinionated.